File: API/Entities/Recipe.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using API.DTO;

namespace API.Entities;

 public class Recipe
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public int CookingTime { get; set; } = 0;
        public ICollection<Photo> Photos { get; set; } = [];
        public float Rating { get; set; } = 0;
        public int? UserId { get; set; }
        public ApplicationUser? User { get; set; }
        public RecipeDifficulty Difficulty { get; set; } = RecipeDifficulty.Easy;
        public ServingType ServingType { get; set; } = null!;
        public DateTime DateAdded { get; set; } = DateTime.UtcNow;
        public ICollection<Allergy> Allergies { get; set; } = [];
        public ICollection<RecipeCookware> Cookware { get; set; } = [];
        public ICollection<RecipeInstruction> Instructions { get; set; } = [];
        public ICollection<RecipeIngredient> Ingredients { get; set; } = [];
    }

File: client/src/app/_models/recipe.ts
  import { Allergy } from "./allergy"
  import { Cookware } from "./cookware"
  import { Photo } from "./photo"
  import { RecipeDifficulty } from "./recipeDifficulty"
  import { RecipeIngredient } from "./recipeIngredient"
  import { RecipeInstruction } from "./recipeInstruction"
  import { ServingType } from "./servingType"

  export interface Recipe {
    id : number
    name : string
    description : string
    cookingTime : number
    photoUrl : string
    rating: number
    Difficulty: RecipeDifficulty
    servingType: ServingType
    dateAdded: Date
    allergies: Allergy[]
    cookware: Cookware[]
    recipeInstructions: RecipeInstruction[]
    recipeIngredients: RecipeIngredient[]

  }
